scope.py

```diff
@@ -839,6 +840,175 @@ def create_inference_script(
+        min_class_examples: int = 1000,
+        probability_threshold: float = 0.9,
+        al_directory: str = 'AL_datasets',
+        filename: str = 'active_learning_set',
+        algorithm: str = 'dnn',
+        exclude_training_sources: bool = False,
+        write_csv: bool = True,
+        verbose: bool = False,
+    ):
+        """
+        Select subset of predictions to use for active learning.
+
+        :param fields: list of field predictions (integers) to include, or 'all' to use all available fields (list or str)
+        :param group: name of group containing trained models within models directory (str)
+        :param min_class_examples: minimum number of examples to include for each class. Some classes may contain fewer than this if the sample is limited (int)
+        :param probability_threshold: minimum probability to select examples for active learning (float)
+        :param al_directory: name of directory to create/populate with active learning sample (str)
+        :param filename: name of file (no extension) to store active learning sample (str)
+        :param algorithm: algorithm [dnn or xgb] (str)
+        :param exclude_training_sources: if True, exclude sources in current training set from AL sample (bool)
+        :param write_csv: if True, write CSV file in addition to HDF5 (bool)
+        :param verbose: if True, print additional information (bool)
+
+        :return:
+
+        :example:  ./scope.py select_al_sample --fields=[296, 297] --group='experiment' --min_class_examples=1000 --probability_threshold=0.9 --exclude_training_sources
+        """
+        base_path = pathlib.Path(__file__).parent.absolute()
+        preds_path = base_path / 'preds'
+
+        if algorithm in ['dnn', 'DNN']:
+            algorithm = 'dnn'
+        elif algorithm in ['xgb', 'XGB']:
+            algorithm = 'xgb'
+        else:
+            raise ValueError('Algorithm must be either dnn or xgb.')
+
+        df_coll = []
+        if fields in ['all', 'All', 'ALL']:
+            gen_fields = os.walk(preds_path)
+            fields = [x for x in gen_fields][0][1]
+        else:
+            fields = [f'field_{f}' for f in fields]
+
+        print(f'Generating active learning sample from {len(fields)} fields.')
+
+        column_nums = []
+        for field in fields:
+            h = read_hdf(str(preds_path / field / f'{field}.h5'))
+            column_nums += [len(h.columns)]
+            df_coll += [h]
+
+            if verbose:
+                print(field)
+                print(h)
+                print()
+
+        if len(np.unique(column_nums)) > 1:
+            raise ValueError('Not all predictions have the same number of columns.')
+
+        preds_df = pd.concat(df_coll, axis=0)
+        # Define non-variable class as 1 - variable
+        include_nonvar = False
+        if f'vnv_{algorithm}' in preds_df.columns:
+            include_nonvar = True
+            preds_df[f'nonvar_{algorithm}'] = np.round(
```

Comment:
Why the rounding?

Reply:
Currently in inference.py all classification probabilities are rounded to 2 decimal places. The same is done here to the non-variable column for consistency with the other classifications.

Reply:
Ah gotcha. Do you understand why you do this?

Reply:
I think the rounding helps with readability when scanning classifications. The rounded probabilities are numerically close to the original values, but when looking at many of them together, a person can make quicker sense of the rounded values.